src/game.cpp

```diff
@@ -885,40 +885,71 @@ class GameOnDemandSoundFetcher: public OnDemandSoundFetcher
 	}
 };
 
+
+// before 1.8 there isn't a "integer interface", only float
+#if (IRRLICHT_VERSION_MAJOR == 1 && IRRLICHT_VERSION_MINOR < 8)
+typedef irr::f32 SamplerLayer_t;
```

Comment:
There is no need to use the `irr` namespace here, its always implicit (see `irrlichttypes.h`). I don't know why old code did this. Same for the else case.